### ISSUE TITLE
[DEV] kernel parameters "console=tty0 console=ttyS0" are not passed to the installed Ubuntu 16.04 LTS node

### ISSUE THREAD

**Issue by immarvin** [xcat2/xcat-core#731]
From @ligc 

I am trying the Ubuntu 16.04 pre-GA build with xCAT, and found that the kernel parameters "console=tty0 console=ttyS0" are not passed to the running system, I have to manually edit /etc/default/grub and add the kernel parameters to make the rcons work.  You may want to add this into your TODO list for Ubuntu 16.04 support.

**Comment by whowutwut** [comment]
@pdlun92 FYI

**Comment by immarvin** [comment]
hi @whowutwut , is @pdlun92 looking into ubuntu 16.04 porting? if yes, you can assign the issue to him

**Comment by daniceexi** [comment]
@neo954 Please verify whether this is a valid bug for Ubuntu 16.04

**Comment by daniceexi** [comment]
@whowutwut You have done the ubuntu 16.04 installation, did the rcons work well? If kernel parameter **console=xx** has problem to be passed to kernel, the rcons won't work.

**Comment by neo954** [comment]
Working on it.

**Comment by neo954** [comment]
Played with Ubuntu 16.04 GA edition on a PowerNV 8335-GTA machine. The kernel command arguments ``console=tty0 console=ttyS0`` was not passing to the kernel. But I got ``console=hvc0``, which worked fine.

**Comment by daniceexi** [comment]
@immarvin Base on @neo954 's test against the GA version of Ubuntu16.04 that everything works fine. Could you confirm with @ligc that which platform he encountered this issue?

**Comment by ligc** [comment]
I am seeing this problem on x86 KVM guests, the PowerKVM guests seem ok.

**Comment by immarvin** [comment]
from @ligc , the provision message can be found on console , the reboot messages can not be seen on console

**Comment by immarvin** [comment]
The workaround is to add "console=tty0 console=ttyS0" into addkcmdline with 
````
chdef <node> addkcmdline="R::console=tty0 R::console=ttyS0"
````
then run nodeset/rpower...

**Comment by daniceexi** [comment]
@immarvin Have you verified this workaround? Should it be **tty0** or **hvc0**?

**Comment by daniceexi** [comment]
Even we had a workaround, will take a look this issue deeply in 2.12.1 to see possible fix.

**Comment by whowutwut** [comment]
👍

**Comment by immarvin** [comment]
fixed in https://github.com/xcat2/xcat-core/pull/1455

**Comment by immarvin** [comment]
verified the fix on ubuntu 16.04 x86_64 kvm guest
